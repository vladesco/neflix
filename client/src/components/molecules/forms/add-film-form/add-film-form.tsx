import React from 'react'
import { Form, Formik } from 'formik'

import { withUsageFormik } from '../../../../hoc'
import { CreateFilmBody } from '../../../../store'
import { Card, TextInput, ThemeButton } from '../../../atoms'
import { Select } from '../../../atoms'
import { GENRES } from '../../../const'
import { addFilmFormStyles } from './add-film-form.style'
import { filmValidationSchema } from '../consts'

type props = {
    onCloseForm: () => void
    onAddFilm: (filmBody: CreateFilmBody) => void
}

const TextInputWithFormik = withUsageFormik(TextInput)
const SelectWithFormik = withUsageFormik(Select)

const defaultFilmModel = {
    title: '',
    poster_path: '',
    overview: '',
    runtime: 0,
    genres: [],
    release_date: '',
}

export const AddFilmForm: React.FC<props> = ({ onCloseForm, onAddFilm }) => {
    const classes = addFilmFormStyles()
    return (
        <Card>
            <div className={classes.form}>
                <span onClick={onCloseForm} className={classes.cross}>
                    &#215;
                </span>
                <div className={classes.content}>
                    <p className={classes.title}>Add movie</p>

                    <Formik
                        initialValues={defaultFilmModel}
                        validationSchema={filmValidationSchema}
                        onSubmit={(values, { setSubmitting }) => {
                            onAddFilm(values)
                            setSubmitting(false)
                        }}
                    >
                        <Form className={classes.form}>
                            <TextInputWithFormik
                                label="Title"
                                name="title"
                                placeholder="Tile here"
                            />

                            <TextInputWithFormik
                                type="date"
                                label="Release Date"
                                name="release_date"
                                placeholder="Select Date"
                            />

                            <TextInputWithFormik
                                label="Movie Url"
                                name="poster_path"
                                placeholder="Movie URL here"
                            />

                            <SelectWithFormik
                                label="Genre"
                                name="genres"
                                className={classes.select}
                                placeholder="Select Genre"
                                items={GENRES}
                                multiple={true}
                            />

                            <TextInputWithFormik
                                label="Overview"
                                name="overview"
                                placeholder="Overview here"
                            />

                            <TextInputWithFormik
                                label="Runtime"
                                name="runtime"
                                type="number"
                                placeholder="Runtime here"
                            />

                            <div className={classes.buttonArea}>
                                <ThemeButton type="reset" theme="DARK">
                                    Reset
                                </ThemeButton>

                                <ThemeButton type="submit" theme="LIGHT">
                                    Submit
                                </ThemeButton>
                            </div>
                        </Form>
                    </Formik>
                </div>
            </div>
        </Card>
    )
}
