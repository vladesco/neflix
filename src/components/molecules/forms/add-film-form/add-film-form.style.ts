import { createUseStyles } from 'react-jss'

export const addFilmFormStyles = createUseStyles({
    form: {
        display: 'flex',
        flexDirection: 'column',
        gap: '16px',
        width: '500px',
    },
    content: {
        display: 'flex',
        flexDirection: 'column',
        gap: '16px',
        '& label': {
            fontSize: '20px',
            textTransform: 'uppercase',
            color: '#F65261',
        },
        '& .error': {
            fontSize: '20px',
            color: '#eb4c2d',
        },
    },
    title: {
        fontSize: '40px',
        color: 'white',
        textTransform: 'uppercase',
    },
    cross: {
        color: 'white',
        fontSize: '48px',
        userSelect: 'none',
        cursor: 'pointer',
        textAlign: 'end',
    },
    buttonArea: {
        display: 'flex',
        justifyContent: 'flex-end',
        marginBottom: '32px',
        gap: '16px',
    },
    select: {
        backgroundColor: '#424242',
        width: '100%',
        color: 'white',
        borderRadius: '5px',
    },
})
