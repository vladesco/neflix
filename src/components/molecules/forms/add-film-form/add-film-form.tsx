import React from 'react'
import { Form, Formik } from 'formik'
import * as Yup from 'yup'

import { withUsageFormik } from '../../../../hoc'
import { CreateFilmBody } from '../../../../store'
import { Card, DateInput, TextInput, ThemeButton } from '../../../atoms'
import { Select } from '../../../atoms'
import { GENRES } from '../../../const'
import { addFilmFormStyles } from './add-film-form.style'
import { filmValidationSchema } from '../consts'

type props = {
    onCloseForm: () => void
    onAddFilm: (filmBody: CreateFilmBody) => void
}

const TextInputWithFormik = withUsageFormik(TextInput)
const DateInputWithFormik = withUsageFormik(DateInput)
const SelectWithFormik = withUsageFormik(Select)

const defaultFilmModel = {
    title: '',
    poster_path: '',
    overview: '',
    runtime: null,
    genres: [],
    release_date: null,
}

export const AddFilmForm: React.FC<props> = ({ onCloseForm, onAddFilm }) => {
    const classes = addFilmFormStyles()
    return (
        <Card>
            <div className={classes.form}>
                <span onClick={onCloseForm} className={classes.cross}>
                    &#215;
                </span>
                <div className={classes.content}>
                    <p className={classes.title}>Add movie</p>

                    <Formik
                        initialValues={defaultFilmModel}
                        validationSchema={filmValidationSchema}
                        onSubmit={(values, { setSubmitting }) => {
                            onAddFilm(values)
                            setSubmitting(false)
                        }}
                    >
                        <Form className={classes.form}>
                            <TextInputWithFormik
                                label="Title"
                                name="title"
                                placeHolder="Tile here"
                            />

                            <DateInputWithFormik
                                label="Release Date"
                                name="release_date"
                                placeholder="Select Date"
                            />

                            <TextInputWithFormik
                                label="Movie Url"
                                name="poster_path"
                                placeHolder="Movie URL here"
                            />

                            <SelectWithFormik
                                label="Genre"
                                name="genres"
                                className={classes.select}
                                placeholder="Select Genre"
                                items={GENRES}
                                multiple={true}
                            />

                            <TextInputWithFormik
                                label="Overview"
                                name="overview"
                                placeHolder="Overview here"
                            />

                            <TextInputWithFormik
                                label="Runtime"
                                name="runtime"
                                type="number"
                                placeHolder="Runtime here"
                            />

                            <div className={classes.buttonArea}>
                                <ThemeButton type="reset" theme="DARK">
                                    Reset
                                </ThemeButton>

                                <ThemeButton type="submit" theme="LIGHT">
                                    Submit
                                </ThemeButton>
                            </div>
                        </Form>
                    </Formik>
                </div>
            </div>
        </Card>
    )
}
