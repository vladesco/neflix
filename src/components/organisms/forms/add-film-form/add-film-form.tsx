import React from 'react'
import { DateInput, Input, ThemeButton } from '../../../atoms'
import { Select } from '../../../atoms'
import { GENRES } from '../../../const'
import { Card } from '../../../molecules'
import { addFilmFormStyles } from './add-film-form.style'

type props = {
    onCloseForm: () => void
    onAddFilm: (filmId: string) => void
}

export const AddFilmForm: React.FC<props> = ({ onCloseForm }) => {
    const classes = addFilmFormStyles()
    return (
        <Card>
            <div className={classes.form}>
                <div className={classes.content}>
                    <p className={classes.title}>Add movie</p>

                    <span className={classes.label}>Title</span>
                    <Input placeHolder="Tile here"></Input>

                    <span className={classes.label}>Release Date</span>
                    <DateInput placeHolder="Select Date" />

                    <span className={classes.label}>Movie URL</span>
                    <Input placeHolder="Movie URL here"></Input>

                    <span className={classes.label}>Genre</span>
                    <Select
                        className={classes.select}
                        placeholder="Select Genre"
                        onSelect={console.log}
                        items={GENRES}
                    ></Select>

                    <span className={classes.label}>Overview</span>
                    <Input placeHolder="Overview here"></Input>

                    <span className={classes.label}>Runtime</span>
                    <Input placeHolder="Runtime here"></Input>

                    <div className={classes.buttonArea}>
                        <ThemeButton theme="DARK">Reset</ThemeButton>
                        <ThemeButton theme="LIGHT">Confirm</ThemeButton>
                    </div>
                </div>
                <span className={classes.cross} onClick={onCloseForm}>
                    &#215;
                </span>
            </div>
        </Card>
    )
}
