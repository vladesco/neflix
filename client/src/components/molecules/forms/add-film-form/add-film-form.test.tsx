import React from 'react'
import { render, screen } from '@testing-library/react'
import { AddFilmForm } from './add-film-form'

describe('add film form', () => {
    it('should be rendered', () => {
        const { asFragment } = render(
            <AddFilmForm onAddFilm={null} onCloseForm={null} />
        )

        expect(asFragment()).toMatchSnapshot()
        expect(screen.getByText('Add movie')).toBeTruthy()
    })
})