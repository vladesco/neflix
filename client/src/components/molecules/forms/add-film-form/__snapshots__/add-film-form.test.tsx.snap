// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add film form should be rendered 1`] = `
<DocumentFragment>
  <div
    class="card-0-2-7"
  >
    <div
      class="form-0-2-1"
    >
      <span
        class="cross-0-2-4"
      >
        Ã—
      </span>
      <div
        class="content-0-2-2"
      >
        <p
          class="title-0-2-3"
        >
          Add movie
        </p>
        <form
          action="#"
          class="form-0-2-1"
        >
          <label
            for="title"
          >
            Title
          </label>
          <input
            class="input-0-2-8"
            name="title"
            placeholder="Tile here"
            value=""
          />
          <label
            for="release_date"
          >
            Release Date
          </label>
          <input
            class="input-0-2-9"
            name="release_date"
            placeholder="Select Date"
            type="date"
            value=""
          />
          <label
            for="poster_path"
          >
            Movie Url
          </label>
          <input
            class="input-0-2-8"
            name="poster_path"
            placeholder="Movie URL here"
            value=""
          />
          <label
            for="genres"
          >
            Genre
          </label>
          <div
            class="wrapper-0-2-10 select-0-2-6"
          >
            <div
              class="select-0-2-11 closed"
            >
              Select Genre
            </div>
          </div>
          <label
            for="overview"
          >
            Overview
          </label>
          <input
            class="input-0-2-8"
            name="overview"
            placeholder="Overview here"
            value=""
          />
          <label
            for="runtime"
          >
            Runtime
          </label>
          <input
            class="input-0-2-8"
            name="runtime"
            placeholder="Runtime here"
            type="number"
            value="0"
          />
          <div
            class="buttonArea-0-2-5"
          >
            <button
              class="button-0-2-14 button-d0-0-2-15"
              type="reset"
            >
              Reset
            </button>
            <button
              class="button-0-2-14 button-d1-0-2-16"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
